example_scenarios/c_client/c_client.cpp

```diff
@@ -0,0 +1,345 @@
+      OutputFoldername + "_" + AppName + "_output_image.pdf";
+
+  json HMDOE;
+  HMDOE["doe_type"] = "standard latin hypercube"; // "random sampling";
+  HMDOE["number_of_samples"] = NumDSERandomSamples;
+
+  HMScenario["design_of_experiment"] = HMDOE;
+
+  for (auto InParam : InParams) {
+    json HMParam;
+    HMParam["parameter_type"] = getTypeAsString(InParam->getType());
+    switch (InParam->getType()) {
+    case Ordinal:
+    case Categorical:
+    case Integer:
+      HMParam["values"] = json(InParam->getRange());
+      break;
+    default:
+      fatalError("Only Ordinal and Categorical handled!");
+      break;
+    }
+    HMScenario["input_parameters"][InParam->getKey()] = HMParam;
+  }
+
+  //  cout << setw(4) << HMScenario << endl;
+  ofstream HyperMapperScenarioFile;
+
+  string JSonFileNameStr =
+      CurrentDir + "/" + OutputFoldername + "/" + AppName + "_scenario.json";
+
+  HyperMapperScenarioFile.open(JSonFileNameStr);
+  if (HyperMapperScenarioFile.fail()) {
+    fatalError("Unable to open file: " + JSonFileNameStr);
+  }
+  cout << "Writing JSON file to: " << JSonFileNameStr << endl;
+  HyperMapperScenarioFile << setw(4) << HMScenario << endl;
+  return JSonFileNameStr;
+}
+
+// Function that takes input parameters and generates objective
+HMObjective calculateObjective(vector<HMInputParam *> &InputParams) {
+
+  HMObjective Obj;
+  int x1 = InputParams[0]->getVal();
+  int x2 = InputParams[1]->getVal();
+
+  Obj.f1_value = 2 + (x1 - 2) * (x1 - 2) + (x2 - 1) * (x2 - 1);
+  Obj.f2_value = 9 * x1 - (x2 - 1) * (x2 - 1);
+
+  bool c1 = ((x1 * x1 + x2 * x2) <= 255);
+  bool c2 = ((x1 - 3 * x2 + 10) <= 0);
+  Obj.valid = c1 && c2;
+  return Obj;
+}
+
+// Function that populates input parameters
+int collectInputParams(vector<HMInputParam *> &InParams) {
+  int numParams = 0;
+
+  vector<int> Range = {-20, 20};
+
+  HMInputParam *x0Param = new HMInputParam("x0", ParamType::Integer);
+  x0Param->setRange(Range);
+  InParams.push_back(x0Param);
+  numParams++;
+
+  HMInputParam *x1Param = new HMInputParam("x1", ParamType::Integer);
+  x1Param->setRange(Range);
+  InParams.push_back(x1Param);
+  numParams++;
+  return numParams;
+}
+
+// Function for mapping input parameter based on key
+auto findHMParamByKey(vector<HMInputParam *> &InParams, string Key) {
+  for (auto it = InParams.begin(); it != InParams.end(); ++it) {
+    HMInputParam Param = **it;
+    if (Param == Key) {
+      return it;
+    }
+  }
+  return InParams.end();
+}
+
+int main(int argc, char **argv) {
+
+  if (!getenv("HYPERMAPPER_HOME") || !getenv("PYTHONPATH")) {
+    string ErrMsg = "Environment variables are not set!\n";
+    ErrMsg += "Please set HYPERMAPPER_HOME and PYTHONPATH before running this ";
+    fatalError(ErrMsg);
+  }
+
+  srand(0);
+
+  vector<HMInputParam *> InParams;
+  
+  // Set these values accordingly
+  // TODO: make these command line inputs
+  string OutputFoldername = "outdata";
+  string AppName = "test";
```

Comment:
Can we use a different name here? Like "c_client_chakong_haimes" or simply "chakong_haimes"?